healsparse/healSparseMap.py

```diff
@@ -350,33 +370,69 @@ def coverageMask(self):
         covMask = (self._covIndexMap[:] + np.arange(hp.nside2npix(self._nsideCoverage))*nfine) >= nfine
         return covMask
 
-    def generateHealpixMap(self, nside=None, reduction='mean'):
+    def generateHealpixMap(self, nside=None, reduction='mean', key=None):
         """
         Generate the associated healpix map
 
         if nside is specified, then reduce
+
+        Args:
+        -----
+
         """
+        if nside is None:
+            nside = self._nsideSparse
+        # Check if we are dealing with recArrays and try to preserve the data types
+        if self._isRecArray:
+            if key is None:
+                raise ValueError('key should be specified for HealSparseMaps including `recarray`')
+            else:
+                if isinstance(self._sparseMap[key],int):
+                    aux_dtype = np.float #Force to float
+                else:
+                    aux_dtype = self._sparseMap[key].dtype
+                aux = self._sparseMap[key].astype(aux_dtype)
+        # If it is not a recArray then, try to preserve the type anyway
+        else:
+            if isinstance(self._sparseMap,int):
+                aux_dtype = np.float #Force to float
+            else:
+                aux_dtype = self._sparseMap.dtype
+            aux = self._sparseMap.astype(aux_dtype)
+        # Create empty HEALPix map
+        hp_map = np.zeros(hp.nside2npix(nside), dtype=aux_dtype) + hp.UNSEEN # remember that empty means UNSEEN!
+        # We generate a HealSparseMap with the requested size before converting to healpix
+        aux_spmap = HealSparseMap(covIndexMap=self._covIndexMap, sparseMap=aux, nsideSparse=self._nsideSparse,
+                                  nsideCoverage=self._nsideCoverage)
```

Comment:
I believe this is only necessary if (a) we are going to degrade the resolution or (b) we have a datatype change (from int to float).  Otherwise, I worry that this uses extra memory.

Reply:
That's a fair comment. I was thinking that, in most cases, this would be a "small" overhead but, I can see that at some point, this is not going to be the case anymore so I changed this. Please, let me know if the changes make sense.